src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as vscode from "vscode";
+import * as nls from "vscode-nls";
+import Sinon, { stub, assert, spy } from "sinon";
+import { expect } from "chai";
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import powerPlatformExtensionContext from "../../../client/WebExtensionContext";
+import { schemaKey } from "../../schema/constants";
+import { telemetryEventNames } from "../../telemetry/constants";
+
+import {
+    queryParameters,
+    PORTALS_FOLDER_NAME_DEFAULT,
+} from "../../common/constants";
+
+import {
+    showErrorDialog,
+    removeEncodingFromParameters,
+    checkMandatoryParameters,
+    checkMandatoryPathParameters,
+    checkMandatoryQueryParameters,
+} from "../../common/errorHandler";
+describe("errorHandler", () => {
+    afterEach(() => {
+        Sinon.restore();
+    });
+    it("showErrorDialog_shouldCallShowErrorMessage", () => {
+        //Act
+        const errorString = "this is error message";
+        const detailMessage = "not able to open";
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+
+        //Action
+        showErrorDialog(errorString, detailMessage);
+
+        //Assert
+        const options = { detail: detailMessage, modal: true };
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+    });
+
+    it("showErrorDialog_shouldCallShowErrorMessage_whenDetailMessagesIsNotPassed", () => {
+        //Act
+        const errorString = "this is error message";
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+
+        //Action
+        showErrorDialog(errorString);
+
+        //Assert
+        const options = { detail: undefined, modal: true };
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+    });
+
+    it("removeEncodingFromParameters_shouldDecodeSCHEMAVERSIONAndWEBSITENAME", () => {
+        //Act
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, "3A%20test_file_name"],
```

Comment:
Why we need 3a? In URL %3a means '/'.
Remove this 3a and keep only test_file_name and test_test_websiteName

Reply:
I am using this because decode URI should remove this 3A from Url and return the decoded url.

Reply:
If we remove that 3A it will return same string as without 3A